context/app/static/js/components/Search/TilesSortDropdown/TilesSortDropdown.jsx

```diff
@@ -1,78 +1,48 @@
-import React, { useRef, useState } from 'react';
+import React from 'react';
 import PropTypes from 'prop-types';
-import Paper from '@material-ui/core/Paper';
-import Popper from '@material-ui/core/Popper';
-import ArrowDropDownIcon from '@material-ui/icons/ArrowDropDownRounded';
-import ArrowDropUpIcon from '@material-ui/icons/ArrowDropUpRounded';
-import MenuList from '@material-ui/core/MenuList';
-import ClickAwayListener from '@material-ui/core/ClickAwayListener';
 
+import DropdownListbox from 'js/shared-styles/dropdowns/DropdownListbox';
 import useSearchViewStore from 'js/stores/useSearchViewStore';
-import { SelectionButton, StyledDropdownSelectItem } from './style';
+import { StyledButton, StyledDropdownListboxOption } from './style';
 import { getSortPairs } from '../utils';
 
 const searchViewStoreSelector = (state) => state.searchView;
 
-function getSelectedItemLabel(items, selectedItems) {
+function getSelectedItemIndex(items, selectedItems) {
   if (selectedItems.length > 1) {
     console.warn('Expected only a single sort, not:', selectedItems);
   }
   const selectedItem = selectedItems.length ? selectedItems[0] : undefined;
-  const match = items.filter((item) => item.key === selectedItem);
-  return match.length ? match[0].label : '';
+  return items.findIndex((item) => [item[0].key, item[1].key].includes(selectedItem)) || 0;
```

Comment:
There shouldn't ever be a case where the selected item isn't found. How would you suggest we handle possible errors? Add a console.warn?

Reply:
I guess I'm stuck on the `|| 0`... What is it there for? Reading [this](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/findIndex), `findIndex` always returns a number, so it would only be false-y if it were already returning 0?